src/test/java/ca/utoronto/fitbook/unit/UpvotePostsServiceTest.java

```diff
@@ -0,0 +1,89 @@
+package ca.utoronto.fitbook.unit;
+
+import ca.utoronto.fitbook.BaseTest;
+import ca.utoronto.fitbook.adapter.persistence.firebase.UserFirebaseRepository;
+import ca.utoronto.fitbook.adapter.persistence.localmemory.PostLocalMemoryRepository;
+import ca.utoronto.fitbook.adapter.persistence.localmemory.UserLocalMemoryRepository;
+import ca.utoronto.fitbook.application.port.in.UpvotePostsUsecase;
+import ca.utoronto.fitbook.application.port.in.command.UpvotePostsCommand;
+import ca.utoronto.fitbook.application.service.UpvotePostsService;
+import ca.utoronto.fitbook.entity.User;
+import org.junit.jupiter.api.*;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+
+import java.util.*;
+
+public class UpvotePostsServiceTest extends BaseTest {
+
+    @Autowired
+    private UserLocalMemoryRepository userLocalMemoryRepository;
+    private PostLocalMemoryRepository postLocalMemoryRepository;
+    private UpvotePostsUsecase upvotePostsUsecase;
+    private User testUser1;
+    private User testUser2;
+
+    @BeforeAll
+    public void init() {
+        this.userLocalMemoryRepository = new UserLocalMemoryRepository();
+        this.postLocalMemoryRepository = new PostLocalMemoryRepository();
+        this.upvotePostsUsecase = new UpvotePostsService(postLocalMemoryRepository, userLocalMemoryRepository);
+
+        testUser1 = User.builder()
+                .followersIdList(new ArrayList<>())
+                .joinDate(new Date())
+                .name("Jan")
+                .totalLikes(1)
+                .password("pw")
+                .postIdList(new ArrayList<>())
+                .followingIdList(new ArrayList<>())
+                .followersIdList(new ArrayList<>())
+                .likedPostIdList(new ArrayList<>())
+                .build();
+        testUser2 = User.builder()
+                .followersIdList(new ArrayList<>())
+                .joinDate(new Date())
+                .name("Feb")
+                .totalLikes(1)
+                .password("pw2")
+                .postIdList(new ArrayList<>())
+                .followingIdList(new ArrayList<>())
+                .followersIdList(new ArrayList<>())
+                .likedPostIdList(new ArrayList<>())
+                .build();
```

Comment:
Using the random user method in TestUtilities should make this a bit cleaner.

Reply:
random user method has been added and previous code removed